inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/management/DynamicWorkloadManagementPage.java

```diff
@@ -342,33 +356,253 @@ protected void onUpdate(AjaxRequestTarget ajaxRequestTarget) {
         unused.setOutputMarkupId(true);
         searchForm.add(unused);
 
-        //Reset button
-        Button reset = new LambdaAjaxButton("reset", this::actionReset);
+        // Reset button
+        Button reset = new LambdaAjaxButton<>("reset", this::actionReset);
 
         searchForm.add(reset);
 
-        //Submit button
-        Button search = new LambdaAjaxButton("search", this::actionSubmit);
+        // Submit button
+        Button search = new LambdaAjaxButton<>("search", this::actionSubmit);
 
         searchForm.add(search);
 
-        //Condition for filter inputs to be enabled
-        dateTo.add(enabledWhen(() ->
-            !dateChoices.getValue().equals(dateChoice.get(0))
-            && unused.getValue().equals("false")));
-        dateFrom.add(enabledWhen(() ->
-            !dateChoices.getValue().equals(dateChoice.get(1))
-            && unused.getValue().equals("false")));
-        dateChoices.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
-        userFilterTextField.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
-        documentFilterTextField.add(enabledWhen(() ->
-            unused.getValue().equals("false")));
+        // Condition for filter inputs to be enabled
+        dateTo.add(enabledWhen(
+            () -> !dateChoices.getDefaultModelObjectAsString().equals(dateChoice.get(0))
+                 && unused.getValue().equals("false")));
```

Comment:
Comparing a boolean value against the string value "false" seems like a terrible hack. There must be a better approach to this.